server/container/util/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -123,6 +134,17 @@ private FluentFutureStream(CompletableFuture<Stream<T>> completableFuture) {
             CompletableFutureUtil.thenComposeOnAll(completableFuture(), function));
     }
 
+    /**
+     * Apply a transformation to all value of the underlying stream.
+     *
+     * As the supplied transformation produces fluent futures of stream, we need to compose then flatMap the returned values.
+     */
+    public <U> FluentFutureStream<U> thenFlatMap(Function<T, FluentFutureStream<U>> function) {
```

Comment:
what about naming it simply flatMap ?

Reply:
I did try hard to make things more composable here: https://github.com/linagora/james-project/pull/1740